### DISCUSSION TITLE
Remove lang attribute from html element or allow removal

### DISCUSSION THREAD

**Discussion by drahkrub** [quasarframework/quasar#14337] [Ideas / Proposals]
If all Quasar default texts are overridden using e.g. vue-i18n than there is no need for the Quasar language packs (or at least for different ones).

Nevertheless the lang attribute is always set by Quasar to "en-us" (default) or some other value.

This is problematic in a multi-language app, because the value of the lang attribute jumps from absence to "en-us" and finally to the language set in vue-i18n - which may be problematic wrt. automatic browser translations kicking in.

There have been similiar discussions in other projects, e.g.

1. https://github.com/vuejs/vue-cli/issues/5945#issuecomment-708356918
2. https://github.com/h5bp/html5-boilerplate/issues/1542
3. https://github.com/h5bp/html5-boilerplate/issues/1879

So it would be nice to create a way to prevent Quasar from touching the lang attribute at all.

**Comment by pdanpdan** [comment]
I think we need more details about the use case:
- are we talking about SSR or non-SSR
- don't you know the language that the page is displayed in?

In SSR case the page should be delivered with the right lang.
In case of non-SSR the lang attr is the one in index template, and if the language selection is done in a boot file the lang attr should be set ok

I'll do a test on a non-SSR to see how the lang attr changes

**Reply by drahkrub** [reply]
In our case it's a multi-language non-SSR. And yes, the lang attr can be set dynamically as described [here](https://quasar.dev/options/quasar-language-packs#dynamical-non-ssr-). But does that help if the browser language can be dynamically changed? And is it unproblematic to set the "Quasar language" to a language pack which is not included in the build (because it's not needed as all default texts are overridden or a language is choosen which is not supported by Quasar)?

**Comment by pdanpdan** [comment]
I just checked:

- by setting the language in boot file the attr is set directly to that language
- if you set the quasar language to a non-existing one it is enough if you provide an object with `isoName: 'xx'` (and `rtl: true` if needed)

**Reply by drahkrub** [reply]
@pdanpdan Thanks for checking!
> if you set the quasar language to a non-existing one it is enough if you provide an object with [...]

And how should this object be "provided"?

But anyway, quite some configuration is required for something that is not needed at all (Quasar language packs). That's why I still think it would be nice to just be able to tell Quasar "don't set the lang attribute" (of cause the default behaviour should remain the current one :-)).

**Comment by pdanpdan** [comment]
`this.$q.lang.set({ isoName: 'xx', rtl: true })` or `const $q = useQuasar; $q.lang.set({ isoName: 'xx', rtl: true })`

**Reply by drahkrub** [reply]
Thanks again! Ok, last question: If `{ isoName: 'xx', rtl: true }` is set as described, what happens if some default text is not provided (by mistake) - will Quasar use "en-us" as fallback or will an error occur? (I could just try it out, but I'm on the go right now ;-))

**Comment by pdanpdan** [comment]
If the language pack with that isoName is not already in quasar then you'll have no texts in the components.

But you can take a file from here: https://github.com/quasarframework/quasar/tree/dev/ui/lang
make a json with the content translated and with the right isoName/rtl and pass it to `$q.lang.set`
